tools.go

```diff
@@ -1,8 +0,0 @@
-// +build tools
-
-package tools
-
-import (
-	_ "github.com/golang/mock/mockgen"
-	_ "golang.org/x/tools/cmd/stringer"
```

Comment:
With this deleted, what prevents these tools from being removed if you run go mod tidy? Won't this cause the these packages to be removed from the go.mod which in turn causes sage targets to fail?

Reply:
These tools are installed through sage in the .sage/tools repo instead, making them only a build-time dependency and not a runtime dependency. We only use them when building and running test so we don't want them as runtime dependencies.

Reply:
Yes but sage is doing "InstallWithModFile` which reads the version from the dependency in the go.mod file and if these tools are not defined in the go.mod how is that working?

Unless I am missing something I think we need to either change sage to use the latest version (and not depend on the go.mod file) or keep this tools.go

Reply:
Now I see what you mean. It has only worked because we already depend on these packages from other sources. I will re-add this file.